subsys/bluetooth/mesh/large_comp_data_srv.c

```diff
@@ -170,17 +170,28 @@ const struct bt_mesh_model_op _bt_mesh_large_comp_data_srv_op[] = {
 
 static int large_comp_data_srv_init(const struct bt_mesh_model *model)
 {
+	const struct bt_mesh_model *config_srv =
+		bt_mesh_model_find(bt_mesh_model_elem(model), BT_MESH_MODEL_ID_CFG_SRV);
+
 	if (!bt_mesh_model_in_primary(model)) {
 		LOG_ERR("Large Composition Data Server only allowed in primary element");
 		return -EINVAL;
 	}
 
+	if (config_srv == NULL) {
+		return -EINVAL;
+	}
+
```

Comment:
Why sandwich the changes around the `if (!bt_mesh_model_in_primary(model))` check?

Reply:
Btw, since Configuration server already checks that it is instantiated on the primary element, it is enough only to check here that Configuration server is present on the same element. Just a small optimization.